changelogs/unreleased/gh-7014-session-disconnect-peer.md

```diff
@@ -0,0 +1,5 @@
+## bugfix/core
```

Comment:
Wait, the test doesn't pass on my machine. It says in the logs:
```
2022-04-12 00:18:16.201 [82751] main/119/main utils.c:456 E> LuajitError: ...est/box-luatest/gh_7014_session_disconnect_peer_test.lua:16: session.peer(): getpeername() failed
```
Perhaps on a closed descriptor getting the peer name might not work. At least on Mac. Could shutdown work? Get the descriptor, call on it shutdown for both channels via FFI or something, then it should trigger close before the connection is actually dead.

Reply:
Right, it look like `getpeername()` called on a shut down socket fd returns `EINVAL`:

[![image](https://user-images.githubusercontent.com/1114511/162998657-c21a3af3-db9a-4abe-9456-09a1dadd4857.png)](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/getpeername.2.html)

I think that leaves us no option but store the peer address in session meta. I reworked the patch accordingly. Please, take a look.